packages/ketcher-core/src/application/editor/modes/SequenceMode.ts

```diff
@@ -835,20 +835,51 @@ export class SequenceMode extends BaseMode {
     const chainsCollection = ChainsCollection.fromMonomers([
       ...drawingEntitiesManager.monomers.values(),
     ]);
+
+    const currentSequence = SequenceRenderer.currentChain;
+
+    const currentSequenceHasPhosphate = currentSequence?.lastNonEmptyNode
+      ? this.hasPhosphateAtEndOfSequence(
+          currentSequence.lastNonEmptyNode.monomer,
+        )
+      : false;
+
+    const newSequenceHasPhosphate = chainsCollection.lastNode
+      ? this.hasPhosphateAtEndOfSequence(chainsCollection.lastNode.monomer)
+      : false;
+
+    let nextCaretPosition =
+      SequenceRenderer.caretPosition + chainsCollection.length;
+
+    if (
+      (currentSequenceHasPhosphate && newSequenceHasPhosphate) ||
+      currentSequenceHasPhosphate
+    ) {
```

Comment:
This condition seems to be either overcomplicated or incorrect for me, it's always `true` if `currentSequenceHasPhosphate` is and vice verse

Reply:
I agree, if you simplify the code here, you don't have to use `hasPhosphateAtEndOfSequence`, thanks!